SwiftReflector/NewClassCompiler.cs

```diff
@@ -6293,8 +6293,10 @@ void SubstituteAssociatedTypeNamer (Func<int, int, string> namer, CSType ty)
 			if (ty is CSGenericReferenceType genType) {
 				genType.ReferenceNamer = namer;
 			} else if (ty is CSSimpleType simpleType) {
-				foreach (var genSubType in simpleType.GenericTypes)
-					SubstituteAssociatedTypeNamer (namer, genSubType);
+				if (simpleType.GenericTypes != null) {
+					foreach (var genSubType in simpleType.GenericTypes)
```

Comment:
I've previously used an extension method for this that returns an empty enumerable, but I'm not sure that is better. Nulls in C# make me sad.

Reply:
That would be nice. I might add that in and forget to use it.
I'm thinking that `GenericTypes` should also be initialized to non-null, but I feel like I had a good reason to have it null some time ago.

Reply:
Here is the copy I used on personal projects:
```
		public static IEnumerable<T> AsNotNull<T> (this IEnumerable<T> original)
		{
			return original ?? Enumerable.Empty<T> ();
		}
```